src/main/scala/com/itechart/ny_accidents/report/generators/ReportGenerators.scala

```diff
@@ -0,0 +1,98 @@
+  def generateReport(data: RDD[_ <: Product], report: Reports): Seq[Seq[String]]
+}
+
+class DayOfWeekReportGenerator extends ReportGenerator {
+  override def calculateReport(data: RDD[MergedData]): RDD[_ <: Product] = {
+    TimeMetricService.countDayOfWeek(data)
+  }
+
+  override def generateReport(data: RDD[_ <: Product], report: Reports): Seq[Seq[String]] = {
+    report.generateReportForTupleRDD(data, DAY_OF_WEEK_REPORT_HEADER)
+  }
+}
+
+class HourOfDayReportGenerator extends ReportGenerator {
+  override def calculateReport(data: RDD[MergedData]): RDD[_ <: Product] = {
+    TimeMetricService.countHours(data)
+  }
+
+  override def generateReport(data: RDD[_ <: Product], report: Reports): Seq[Seq[String]] = {
+    report.generateReportForTupleRDD(data, HOUR_OF_DAY_REPORT_HEADER)
+  }
+}
+
+class PeriodReportGenerator extends ReportGenerator {
+  override def calculateReport(data: RDD[MergedData]): RDD[_ <: Product] = {
+    WeatherMetricService.definePeriod(data)
+  }
+
+  override def generateReport(data: RDD[_ <: Product], report: Reports): Seq[Seq[String]] = {
+    report.generateReportForTupleRDD(data, DAY_OF_WEEK_REPORT_HEADER)
+  }
+}
+
+class WeatherReportGenerator extends ReportGenerator {
+  override def calculateReport(data: RDD[MergedData]): RDD[_ <: Product] = {
+    WeatherMetricService.getPhenomenonPercentage(data)
+  }
+
+  override def generateReport(data: RDD[_ <: Product], report: Reports): Seq[Seq[String]] = {
+    report.generateReportForTupleRDD(data, PHENOMENON_REPORT_HEADER)
+  }
+}
+
+class BoroughReportGenerator extends ReportGenerator {
+  override def calculateReport(data: RDD[MergedData]): RDD[_ <: Product] = {
+    DistrictMetricService.getBoroughPercentage(data)
+  }
+
+  override def generateReport(data: RDD[_ <: Product], report: Reports): Seq[Seq[String]] = {
+    report.generateReportForTupleRDD(data, BOROUGH_REPORT_HEADER)
+  }
+}
+
+class DistrictReportGenerator extends ReportGenerator {
+  override def calculateReport(data: RDD[MergedData]): RDD[_ <: Product] = {
+    DistrictMetricService.getDistrictsPercentage(data)
+  }
+
+  override def generateReport(data: RDD[_ <: Product], report: Reports): Seq[Seq[String]] = {
+    report.generateReportForTupleRDD(data, DISTRICT_REPORT_HEADER)
+  }
+}
+
+class PopulationToNumberOfAccidentsReportGenerator @Inject()(populationService: PopulationMetricService) extends ReportGenerator {
+  override def calculateReport(data: RDD[MergedData]): RDD[_ <: Product] = {
+    populationService.getPopulationToNumberOfAccidentsRatio(data)
+  }
+
+  override def generateReport(data: RDD[_ <: Product], report: Reports): Seq[Seq[String]] = {
+    report.generateReportForTupleRDD(data, DISTRICT_REPORT_HEADER)
+  }
+}
+
+class AccidentCountDuringPhenomenonPerHourReportGenerator extends ReportGenerator {
+  def calculateReport(data: RDD[MergedData], weatherPhenomenon: RDD[(String, Int, Double)]): RDD[_ <: Product] = {
```

Comment:
Seems to be useless method: it just call another service passing the same args, what sense? Why you cannot just call `WeatherMetricService.calculateAccidentCountDuringPhenomenonPerHour` in place instead?